app/src/main/java/com/raghav/spacedawn/di/AppModule.kt

```diff
@@ -0,0 +1,98 @@
+package com.raghav.spacedawn.di
+
+import android.content.Context
+import androidx.room.Room
+import com.raghav.spacedawn.db.ReminderDao
+import com.raghav.spacedawn.db.ReminderDatabase
+import com.raghav.spacedawn.network.LaunchLibrary
+import com.raghav.spacedawn.network.SpaceFlightAPI
+import com.raghav.spacedawn.repository.AppRepository
+import com.raghav.spacedawn.repository.IAppRepository
+import com.raghav.spacedawn.utils.AppApplication
+import com.raghav.spacedawn.utils.Constants
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.android.qualifiers.ApplicationContext
+import dagger.hilt.components.SingletonComponent
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+import javax.inject.Named
+import javax.inject.Singleton
+
+@Module
+@InstallIn(SingletonComponent::class)
+object AppModule {
+
+    @Singleton
+    @Provides
+    fun provideReminderDatabase(
+        @ApplicationContext context: Context
+    ) = Room.databaseBuilder(
+        context,
+        ReminderDatabase::class.java,
+        "reminder_db.db"
+    ).build()
+
+    @Singleton
+    @Provides
+    fun provideReminderDao(database: ReminderDatabase) = database.getRemindersDao()
+
+    @Singleton
+    @Provides
+    fun provideRepository(
```

Comment:
dont think we need to define a provider for repo here since all the dependencies have already been defined with provides

Reply:
ok